zaza/charm_lifecycle/deploy.py

```diff
@@ -118,7 +118,8 @@ def get_jinja2_env():
     """
     template_dir = get_overlay_template_dir()
     return jinja2.Environment(
-        loader=jinja2.FileSystemLoader(template_dir)
+        loader=jinja2.FileSystemLoader(template_dir),
+        undefined=jinja2.StrictUndefined
```

Comment:
Does this have the potential to break assumptions? i.e. some tests don't bother setting things as it's not needed for that scenario?

Reply:
It shouldn't, as we only use this to render jinja variables into bundles, and those are per test anyway?

Reply:
definitely open for discussion, but I'm not entirely certain of other ways to ensure that we require variables that are set in the test bundle's overlay

Reply:
Okay, I'm largely convinced.  Therefore, this is +1 from me.